app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala

```diff
@@ -202,11 +210,19 @@ object RosmProp {
   implicit val format: OFormat[RosmProp] = derived.oformat()
 }
 
-sealed trait UserField
+sealed trait UserField {
+  def fold[B](f: => B)(g: UserField.Enrolment => B)(h: => B): B =
```

Comment:
From caller point of view this seems as unsafe definition. There is nothing to say what `f` vs `h` cases stands for.

Reply:
valid point, so it should be something like

```
def fold[B](f: UserField.AffinityGroup.type => B)(g: UserField.Enrolment => B)(h: UserField.Something.type => B): B =
```

Reply:
Done